pkgs/tools/networking/boundary/default.nix

```diff
@@ -18,9 +18,9 @@ stdenv.mkDerivation rec {
       };
     in
     fetchsrc version {
```

Comment:
This doesn't have to be a function, because it is only invoked for one `version` and one `sha256` argument.

Reply:
The function was originally defined outside of the `stdenv.mkDerivation` until I was asked to move it inside so it needed to be passed version and I thought it was quite readable that way: https://github.com/NixOS/nixpkgs/blob/29bf7e2b89110dfaa0821a432228cdea316cc793/pkgs/tools/networking/boundary/default.nix

I can just remove the version part but I'm not sure what to do for the sha256:

```
src =
    let
      inherit (stdenv.hostPlatform) system;
      suffix = {
        x86_64-linux = "linux_amd64";
        aarch64-linux = "linux_arm64";
        x86_64-darwin = "darwin_amd64";
      }.${system} or (throw "Unsupported system: ${system}");
      fetchsrc = s: fetchzip {
        url = "https://releases.hashicorp.com/boundary/${version}/boundary_${version}_${suffix}.zip";
        sha256 = s.${system};
      };
    in
    fetchsrc {
      x86_64-linux = "sha256-K9WlOS9pRhBcoGaSZ3AhbGoTfKUrn4SDq50x6wnIib8=";
      aarch64-linux = "sha256-BRm0Hy9vuxZSYINrPIMwl2UGZdWTSixLT8EJ1OzCGMg=";
      x86_64-darwin = "sha256-pinmzz6GbQWAj3Ut/IB2o75w8X+CS2tXhE6nekYO4vc=";
    };
```

LMK what you suggest :)

Reply:
```
src =
    let
      inherit (stdenv.hostPlatform) system;
      suffix = {
        x86_64-linux = "linux_amd64";
        aarch64-linux = "linux_arm64";
        x86_64-darwin = "darwin_amd64";
      }.${system} or (throw "Unsupported system: ${system}");
      sha256 = {
        x86_64-linux = "sha256-K9WlOS9pRhBcoGaSZ3AhbGoTfKUrn4SDq50x6wnIib8=";
        aarch64-linux = "sha256-BRm0Hy9vuxZSYINrPIMwl2UGZdWTSixLT8EJ1OzCGMg=";
        x86_64-darwin = "sha256-pinmzz6GbQWAj3Ut/IB2o75w8X+CS2tXhE6nekYO4vc=";
      }.${system} or (throw "Unsupported system: ${system}");
    in
    fetchzip {
      url = "https://releases.hashicorp.com/boundary/${version}/boundary_${version}_${suffix}.zip";
      inherit sha256;
    };
```

Something like this?
I've applied the shas and then inlined `fetchsrc`.
I've also duplicated the error message, so that it doesn't depend on the order of evaluation.

Reply:
ok cool. Thoughts on a function to have the error message twice but not have to type it out both times? Or just leave it as you had it?

```
  src =
    let
      inherit (stdenv.hostPlatform) system;
      access = obj: obj.${system} or (throw "Unsupported system: ${system}");
      suffix = access {
        x86_64-linux = "linux_amd64";
        aarch64-linux = "linux_arm64";
        x86_64-darwin = "darwin_amd64";
      };
      sha256 = access {
        x86_64-linux = "sha256-K9WlOS9pRhBcoGaSZ3AhbGoTfKUrn4SDq50x6wnIib8=";
        aarch64-linux = "sha256-BRm0Hy9vuxZSYINrPIMwl2UGZdWTSixLT8EJ1OzCGMg=";
        x86_64-darwin = "sha256-pinmzz6GbQWAj3Ut/IB2o75w8X+CS2tXhE6nekYO4vc=";
      };
    in
    fetchzip {
      url = "https://releases.hashicorp.com/boundary/${version}/boundary_${version}_${suffix}.zip";
      inherit sha256;
    };
```

Reply:
Sure, why not :)
How about `selectSystem`?
"select" is the technical term for `.` in Nix. https://nixos.org/manual/nix/stable/#table-operators

Reply:
Ok! Would set be a better name than obj? https://nixos.wiki/wiki/Nix_Expression_Language#Types I don't think `set` is a special word but I just want to check :sweat_smile:

Reply:
Shouldn't be a problem, but `attrs` is the convention for "generic attribute set".

Reply:
Thanks for your help and the review :smiley: